internal/app/go-filecoin/internal/submodule/storage_miner_node.go

```diff
@@ -0,0 +1,508 @@
+// the nodes message pool and returns its identity (or an error, if one is
+// encountered).
+func (m *StorageMinerNodeAdapter) SendPreCommitSector(ctx context.Context, sectorID uint64, commR []byte, ticket storage.SealTicket, pieces ...storage.Piece) (cid.Cid, error) {
+	dealIds := make([]types.Uint64, len(pieces))
+	for i, piece := range pieces {
+		dealIds[i] = types.Uint64(piece.DealID)
+	}
+
+	info := &types.SectorPreCommitInfo{
+		SectorNumber: types.Uint64(sectorID),
+
+		CommR:     commR,
+		SealEpoch: types.Uint64(ticket.BlockHeight),
+		DealIDs:   dealIds,
+	}
+
+	precommitParams, err := abi.ToEncodedValues(info)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	mcid, cerr, err := m.messaging.Outbox.Send(
+		ctx,
+		m.workerAddr,
+		m.minerAddr,
+		types.ZeroAttoFIL,
+		types.NewGasPrice(1),
+		types.NewGasUnits(300),
+		true,
+		storagemarket.PreCommitSector,
+		precommitParams,
+	)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	err = <-cerr
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	return mcid, nil
+}
+
+// WaitForPreCommitSector blocks until the pre-commit sector message is mined
+// into a block, returning the block's height and message's exit code (or an
+// error if one is encountered).
+func (m *StorageMinerNodeAdapter) WaitForPreCommitSector(ctx context.Context, mcid cid.Cid) (uint64, uint8, error) {
+	return m.waitForMessageHeight(ctx, mcid)
+}
+
+// SendProveCommitSector creates a commit sector message and places it into the
+// nodes message pool and returns its identity (or an error, if one is
+// encountered).
+func (m *StorageMinerNodeAdapter) SendProveCommitSector(ctx context.Context, sectorID uint64, proof []byte, deals ...uint64) (cid.Cid, error) {
+	dealIds := make([]types.Uint64, len(deals))
+	for i, deal := range deals {
+		dealIds[i] = types.Uint64(deal)
+	}
+
+	info := &types.SectorProveCommitInfo{
+		Proof:    proof,
+		SectorID: types.Uint64(sectorID),
+		DealIDs:  dealIds,
+	}
+
+	commitParams, err := abi.ToEncodedValues(info)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	mcid, cerr, err := m.messaging.Outbox.Send(
+		ctx,
+		m.workerAddr,
+		m.minerAddr,
+		types.ZeroAttoFIL,
+		types.NewGasPrice(1),
+		types.NewGasUnits(300),
+		true,
+		storagemarket.CommitSector,
+		commitParams,
+	)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	err = <-cerr
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	return mcid, nil
+}
+
+// WaitForProveCommitSector blocks until the provided pre-commit message has
+// been mined into the chain, producing the height of the block in which the
+// message was mined (and the message's exit code) or an error if any is
+// encountered.
+func (m *StorageMinerNodeAdapter) WaitForProveCommitSector(ctx context.Context, mcid cid.Cid) (uint64, uint8, error) {
+	return m.waitForMessageHeight(ctx, mcid)
+}
+
+// GetSealTicket produces the seal ticket used when pre-committing a sector at
+// the moment it is called
```

Comment:
@ZenGround0 

> Having a SealTicket reference the precommit and SealSeed reference the provecommit message is confusing.

Hrm, not sure how to make this more clear. A (seal) ticket is used to pre-commit (encode) a sector. A (seal) seed is used to commit a sector. 

I am reluctant to use the terms "pre-commit seed" and "provide commit seed" in the go-filecoin software because those terms do not exist in any of go-sectorbuilder, go-storage-miner, filecoin-ffi, or rust-fil-proofs.

The spec uses names and types which are very different from the above Go or Rust software. The `StorageMinerActor#_rtVerifySealOrAbort` method contains the following:

```
func (a *StorageMinerActor) _rtVerifySealOrAbort(rt Runtime, onChainInfo *abi.OnChainSealVerifyInfo) {
	...

	IMPL_TODO() // Use randomness APIs
	var svInfoRandomness abi.Randomness
	var svInfoInteractiveRandomness abi.Randomness

	svInfo := abi.SealVerifyInfo{
		SectorID: abi.SectorID{
			Miner:  abi.ActorID(minerActorID),
			Number: onChainInfo.SectorNumber,
		},
		OnChain:               *onChainInfo,
		Randomness:            abi.SealRandomness(svInfoRandomness),
		InteractiveRandomness: abi.InteractiveSealRandomness(svInfoInteractiveRandomness),
		UnsealedCID:           unsealedCID,
	}

        ...
}
```

I will confer with @dignifiedquire on this - but I think we should probably rename (in go-filecoin, lotus, go-sectorbuilder and filecoin-ff):

- the randomness-bytes which are passed to pre-commit to "randomness"
- the randomness-bytes which are passed to commit to "randomness" and "interactive randomness"

@ZenGround0 thoughts?

Reply:
Perfect, thanks for communicating this.  100% agree on your suggestion at the end.  Also looks like its out of scope for this PR and will probably trickle up to us through ffi changes when we update.